### PR TITLE
BI-7522 Rework MongoDB company profile service wrapper to use aggregation

### PR DESCRIPTION
**Pull request by dlgroves-ch**
* Create separate MongoDB service wrappers for company profile and
disqualifications.
* Use aggregation query to fetch IDs and company names.
* Aggregate company profile results into Results object.
* Cache MongoDB results in caffeine cache.

### CODE REVIEW

**Review by kinpang-CH — APPROVED** [review]

**Review by cgregoryCH — APPROVED** [review]

File: src/main/java/uk/gov/companieshouse/reconciliation/service/mongo/MongoCompanyProfileCollectionRoute.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    @Override
+    public void configure() throws Exception {
+        from("direct:mongodb-company_profile-collection")
+                .setHeader(CaffeineConstants.ACTION).constant(CaffeineConstants.ACTION_GET)
+                .setHeader(CaffeineConstants.KEY).constant("{{endpoint.mongodb.company_profile.cache.key}}")
+                .to("{{endpoint.cache}}")
+                .choice()
+                .when(header(CaffeineConstants.ACTION_HAS_RESULT).isEqualTo(false))
+                    .setBody().constant(Collections.singletonList(Aggregates.project(Projections.include("_id", "data.company_name"))))
+                    .toD("${header.MongoEndpoint}")
```
Comment (dlgroves-ch) [inline src/main/java/uk/gov/companieshouse/reconciliation/service/mongo/MongoCompanyProfileCollectionRoute.java:33]: TODO: Remove MongoEndpoint header from trigger pipeline and use value of endpoint.mongodb.company_profile_collection in application.properties.

File: src/main/java/uk/gov/companieshouse/reconciliation/service/mongo/MongoCompanyProfileCollectionRoute.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+import com.mongodb.client.model.Projections;
+import org.apache.camel.builder.RouteBuilder;
+import org.apache.camel.component.caffeine.CaffeineConstants;
+import org.springframework.stereotype.Component;
+import uk.gov.companieshouse.reconciliation.function.compare_collection.entity.ResourceList;
+
+import java.util.Collections;
+
+/**
+ * Retrieves and aggregates results from a mongo.collection.distinct query.<br>
```
Comment (dlgroves-ch) [inline src/main/java/uk/gov/companieshouse/reconciliation/service/mongo/MongoCompanyProfileCollectionRoute.java:13]: TODO: Update Javadocs

File: src/main/java/uk/gov/companieshouse/reconciliation/service/mongo/MongoDisqualificationsCollectionRoute.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    @Override
+    public void configure() throws Exception {
+        from("direct:mongodb-disqualifications-collection")
+                .setHeader(CaffeineConstants.ACTION).constant(CaffeineConstants.ACTION_GET)
+                .setHeader(CaffeineConstants.KEY).constant("{{endpoint.mongodb.disqualifications.cache.key}}")
+                .to("{{endpoint.cache}}")
+                .choice()
+                .when(header(CaffeineConstants.ACTION_HAS_RESULT).isEqualTo(false))
+                    .setHeader(MongoDbConstants.DISTINCT_QUERY_FIELD).constant("officer_id_raw")
+                    .toD("${header.MongoEndpoint}")
```
Comment (dlgroves-ch) [inline src/main/java/uk/gov/companieshouse/reconciliation/service/mongo/MongoDisqualificationsCollectionRoute.java:34]: TODO: remove MongoEndpoint header from trigger pipeline and use value of endpoint.mongodb.disqualifications_collection in application.properties.

File: src/main/java/uk/gov/companieshouse/reconciliation/service/mongo/MongoDistinctToResourceListTransformer.java [hunk 4]
```diff
@@ -12,16 +11,13 @@
 import java.util.Set;
 import java.util.stream.Collectors;
 
-@Component
-public class MongoDistinctSelectionTransformer {
+public class MongoDistinctToResourceListTransformer {
```
Comment (dlgroves-ch) [inline src/main/java/uk/gov/companieshouse/reconciliation/service/mongo/MongoDistinctToResourceListTransformer.java:14]: TODO: Javadocs

File: src/main/java/uk/gov/companieshouse/reconciliation/service/transformer/ResultsToCompanyNumberTransformer.java [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+import uk.gov.companieshouse.reconciliation.function.compare_collection.entity.ResourceList;
+import uk.gov.companieshouse.reconciliation.model.ResultModel;
+import uk.gov.companieshouse.reconciliation.model.Results;
+
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@Component
+public class ResultsToCompanyNumberTransformer {
```
Comment (dlgroves-ch) [inline src/main/java/uk/gov/companieshouse/reconciliation/service/transformer/ResultsToCompanyNumberTransformer.java:13]: TODO: Javadocs

**Review by kinpang-CH — APPROVED** [review]

**Review by cgregoryCH — APPROVED** [review]

**Pull request merged**